File: tooltip-action/src/main/java/com/github/kostasdrakonakis/android/animations/FadeAnimation.java
package com.github.kostasdrakonakis.android.animations;

import android.animation.Animator;
import android.view.View;

import com.github.kostasdrakonakis.listeners.TooltipAnimationListener;

public class FadeAnimation implements TooltipAnimationListener {

    private long fadeDuration = 400;

    public FadeAnimation() {
    }

    public FadeAnimation(long fadeDuration) {
        this.fadeDuration = fadeDuration;
    }

    @Override
    public void onAnimationEntered(View view, Animator.AnimatorListener animatorListener) {
        view.setAlpha(0);
        view.animate().alpha(1).setDuration(fadeDuration).setListener(animatorListener);
    }

    @Override
    public void onAnimationExited(View view, Animator.AnimatorListener animatorListener) {
        view.animate().alpha(0).setDuration(fadeDuration).setListener(animatorListener);
    }
}

File: tooltip-action/src/main/java/com/github/kostasdrakonakis/android/animations/FadeAnimation.kt
package com.github.kostasdrakonakis.android.animations

import android.animation.Animator
import android.view.View
import com.github.kostasdrakonakis.listeners.TooltipAnimationListener

class FadeAnimation : TooltipAnimationListener {
    private var fadeDuration: Long = 400

    constructor(): this(400)
    constructor(fadeDuration: Long) {
        this.fadeDuration = fadeDuration
    }

    override fun onAnimationEntered(view: View, animatorListener: Animator.AnimatorListener) {
        view.alpha = 0f
        view.animate().alpha(1f).setDuration(fadeDuration).setListener(animatorListener)
    }

    override fun onAnimationExited(view: View, animatorListener: Animator.AnimatorListener) {
        view.animate().alpha(0f).setDuration(fadeDuration).setListener(animatorListener)
    }
}